src/main/java/seedu/address/logic/commands/DeleteShiftCommand.java

```diff
@@ -0,0 +1,113 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DAY_SHIFT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.time.DayOfWeek;
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Slot;
+import seedu.address.model.person.exceptions.NoShiftException;
+
+/**
+ * Deletes a shift from a staff's schedule.
+ */
+public class DeleteShiftCommand extends Command {
+    public static final String COMMAND_WORD = "deleteShift";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": deletes a shift from the staff identified "
+            + "by\nthe index number used in the displayed staff list or\nthe name of staff." + "\n"
+            + "Parameters: "
+            + "[" + PREFIX_INDEX + "INDEX] or "
+            + "[" + PREFIX_NAME + "NAME] + "
+            + "[" + PREFIX_DAY_SHIFT + "DAY AND\nSLOT OF A SHIFT]" + "\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_INDEX + "1 "
+            + PREFIX_DAY_SHIFT + "monday-1" + "\n"
+            + "OR: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "JOE "
+            + PREFIX_DAY_SHIFT + "TUESDAY-0";
+
+    public static final String MESSAGE_DELETE_SHIFT_SUCCESS = "Shift deleted from the schedule of %s: %s, %s.";
+    public static final String MESSAGE_SHIFT_DOESNT_EXIST = "The shift that you are trying to delete does not exist!";
+
+    private final Index index;
+    private final Name name;
+    private final DayOfWeek dayOfWeek;
+    private final Slot slot;
+
+    /**
+     * Creates a DeleteShiftCommand to add the specified {@code Shift} to a {@code Person}.
+     */
+    public DeleteShiftCommand(Index index, Name name, String shiftDateAndSlot) {
+        requireNonNull(shiftDateAndSlot);
+        this.index = index;
+        this.name = name;
+        String[] strings = shiftDateAndSlot.split("-");
+        dayOfWeek = DayOfWeek.valueOf(strings[0].toUpperCase());
+        slot = Slot.getSlotByOrder(strings[1]);
```

Comment:
Maybe can consider how to deal with the case that the shiftDateAndSlot cannot be divided into a size 2 array?

Reply:
The ParserUtil.parseDayOfWeek() function in the DeleteShiftCommandParser and the AddShiftCommandParser handles this case.